frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -218,6 +211,30 @@ function ReaderFooter:init()
         item_prefix = "icons"
     }
 
+    if not self.settings.order then
+        self.mode_nb = 0
+        self.mode_index = {}
+        local mode_tbl = {}
+        for k,v in pairs(MODE) do
+            mode_tbl[v] = k
+        end
+        local mode_name
+        for i = 0, #mode_tbl do
+            mode_name = mode_tbl[i]
+            if not ((mode_name == "wifi_status" and not Device:isAndroid())
+                or (mode_name == "frontlight" and not Device:hasFrontlight())) then
```

Comment:
That was a bit hard to read :)
A small indentation might help (I like that kind of stuff, dunno if it's ok with our coding style:):
```lua
            if not (   (mode_name == "wifi_status" and not Device:isAndroid())
                    or (mode_name == "frontlight" and not Device:hasFrontlight()) ) then
```
Or clearer:
```lua
           if mode_name == "wifi_status" and not Device:isAndroid() then
               do end -- luacheck: ignore 541
           else if mode_name == "frontlight" and not Device:hasFrontlight() then
               do end -- luacheck: ignore 541
           else
             -- do your suff
           end
```

Reply:
I agree that the second suggestion is clearer, but mainly because it's easier to read the conditions separately than having a `not (something or other)`.

You'd gain the same effect from something like this without the (imo) distracting `do end` stuff.
```lua
            if      not (mode_name == "wifi_status" and not Device:isAndroid())
                and not (mode_name == "frontlight" and not Device:hasFrontlight()) then
```

As an aside, you can also avoid LuaCheck's ire by writing:
```lua
if something then do
    -- No-op.
end
```
Because then it's not empty. ;-)